### ISSUE TITLE
Define explicit capabalities that probe requires

### ISSUE THREAD

**Issue by errordeveloper** [weaveworks/scope#1913]
It'd would be more secure to set explicit capabalities on the probe container, and not require full privileges.

So far, I've managed to launch a probe with `SYS_ADMIN`, `SYS_PTRACE` and `NET_ADMIN`, i.e.:
```
docker run -ti \
  --cap-add=SYS_ADMIN \
  --cap-add=SYS_PTRACE \
  --cap-add=NET_ADMIN \
  --pid=host \
  --net=host \
  --volume=/var/run/docker.sock:/var/run/docker.sock \
    weaveworks/scope:0.17.1
```

This may be enough, but it'd good to know if there are more that we may be aware of. For example, I'm aware that we have a custom DNS resolver, should it need `CAP_NET_RAW` or it was only required for ICMP?

**Comment by errordeveloper** [comment]
See [`capabilities(7)`](http://man7.org/linux/man-pages/man7/capabilities.7.html) for more information.

**Comment by tomwilkie** [comment]
> It'd would be more secure to set explicit capabalities on the probe container, and not require full privileges.

Not true.  Scope requires the docker socket, and therefore can spawn a container for privilege escalation.  Unless we can get a read only docker socket?

**Comment by rade** [comment]
> Scope requires the docker socket, and therefore can spawn a container for privilege escalation.

Unless that action has been restricted via a docker authorization plugin.

**Comment by errordeveloper** [comment]
We can mount the socket with 'ro' flag, but it would break controls, and
probably everything, as you need to write requests to it...

**Comment by 2opremio** [comment]
Closing this as per @tomwilkie's remark

**Comment by rade** [comment]
re-opening. See my comment above - having access to the docker socket does not necessarily provide privilege escalation.

The reason for using more restricted capabilities is that Docker really encourages it and some recent docker features, like swarm mode, do not permit running privileged containers.
